core/services/subscription_internal_test.go

```diff
@@ -42,12 +45,10 @@ func TestSubscriptionListenToLogs(t *testing.T) {
 			err: err,
 			t:   t,
 		},
+		callback: func(manager RunManager, request models.LogRequest) { callbackCalled <- struct{}{} },
 	}
-	done2 := make(chan struct{})
-	go func() {
-		s.Start()
-		done2 <- struct{}{}
-	}()
+	// Note spawns a goroutine
+	s.Start()
```

Comment:
Should this be synchronous or something?

Reply:
no I think its ok as is - all the channels are set up prior to starting and at the end we wait for close(sub.logs). Also fwiw this code will be removed asap